packages/p2p/src/components/floating-rate/floating-rate.jsx

```diff
@@ -0,0 +1,83 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { InputField } from '@deriv/components';
+import { isMobile } from '@deriv/shared';
+import { localize } from 'Components/i18next';
+import { useStores } from 'Stores';
+import './floating-rate.scss';
+
+const FloatingRate = ({
+    changeHandler,
+    class_name,
+    exchange_rate,
+    error_messages,
+    fiat_currency,
+    local_currency,
+    offset,
+    place_holder,
+    ...props
+}) => {
+    const { general_store } = useStores();
+    const { name, value, required } = props;
+    const market_feed = value ? parseFloat(exchange_rate * (1 + value / 100)).toFixed(2) : exchange_rate;
+
+    return (
+        <div className={classNames(class_name, 'floating-rate')}>
+            <section className={classNames('floating-rate__field', isMobile() ? 'mobile-layout' : '')}>
+                <InputField
+                    ariaLabel='Floating rate'
+                    classNameInlinePrefix='floating-rate__percent'
+                    classNameInput={classNames(
+                        'floating-rate__input',
+                        error_messages ? 'floating-rate__input__error_field' : ''
+                    )}
+                    decimal_point_change={2}
+                    id='floating_rate_input'
+                    inline_prefix='%'
+                    is_autocomplete_disabled
+                    is_float
+                    is_incrementable
+                    is_signed
+                    inputmode='decimal'
+                    increment_button_type='button'
+                    max_value={offset.upper_limit}
+                    min_value={offset.lower_limit}
+                    name={name}
+                    onChange={changeHandler}
+                    placeholder={place_holder}
+                    setCurrentFocus={general_store.setCurrentFocus}
```

Comment:
Why do we need this?

Reply:
The Input component has a focus attribute that accepts a function. The function is available in client store. General store has access to the props which contain client store properties. Hence making use of it

Reply:
What happens if it's null?

Reply:
I have removed the fallback to null